### ISSUE TITLE
Internal JSON-RPC error

### ISSUE THREAD

**Issue by johnnyzen** [ethereum/remix-project#3369]
Hi I am getting Internal JSON-RPC error trying to publish a smart contract on mainnet but VM-London is fine?
any ideas on whats happening?
was working the other day, same contract code
I am thinking issue with Remix?

**Comment by Bitcoinpunks** [comment]
Having the same issue

**Comment by Aniket-Engg** [comment]
Hello @johnnyzen @Bitcoinpunks , can you add a bit more details? What do you mean by `publish a smart contract`? Can you add an error screenshot?

**Comment by johnnyzen** [comment]
Sorry, deploying a new contract.

I can deploy on VM-London, but using Brave wallet, I cannot deploy to mainnet.

This is a contract that has previously worked with no changes (prob last tested a week or two ago)

**Comment by johnnyzen** [comment]
I am not getting a full error. It literally just says "Internal JSON-RPC error" in the console output.

**Comment by johnnyzen** [comment]
Also to note, the sample sol contracts on remix, are also failing on being deployed with the same error.

**Comment by johnnyzen** [comment]
![image](https://user-images.githubusercontent.com/203628/218787778-af7a58e7-c737-4c8c-a8db-4b1b383d3d75.png)

**Comment by yann300** [comment]
to which network are you connected too? can you try to deploy to another network?

**Comment by johnnyzen** [comment]
We have tried Polygon mainnet and mumbai Polygon test network. At first we thought it was the test network, so we tried mainnet and that failed too.

**Comment by johnnyzen** [comment]
We are not seeing any reported issues on these networks

**Comment by Bitcoinpunks** [comment]
Have exactly them same. I could just for now test it with Brave.

**Comment by johnnyzen** [comment]
I managed to deploy the contract using metamask/firefox and visual studio code extension. Not sure what issue is, or if its related to brave browser.

**Comment by Bitcoinpunks** [comment]
Thanks Johnny i will try that as well.

**Comment by Bitcoinpunks** [comment]
When i tried to deploy my contract via VScode -> ganache -> wallet connect -> metamask -> goerli, I get the following error message: 
Invalid JSON RPC response: {"id":179,"jsonrpc":"2.0"}

Any ideas?

**Comment by yann300** [comment]
could you sent over the URL you are curently using to connect to the external node?

**Comment by Bitcoinpunks** [comment]
Not sure how to see that since in am connecting it to wallet connect and
meta mask

Or is that the link in the QR of WC?

**Comment by yann300** [comment]
could you try with metamask instead of with the brave wallet?

**Comment by Bitcoinpunks** [comment]
This was the path I went to try it VScode -> ganache -> wallet connect -> metamask -> goerli

I can perfectly deploy it on local though.

**Comment by johnnyzen** [comment]
@Bitcoinpunks have you tried a different rpc? are you using polygon? if so, which RPC are you using?

**Comment by Bitcoinpunks** [comment]
I am using Ethereum and then connect to goerli. Is there a way to connect
to another RPC?

**Comment by johnnyzen** [comment]
Yes, from metamask: https://metamask.zendesk.com/hc/en-us/articles/360043227612-How-to-add-a-custom-network-RPC

Not sure if it will help you, my knowledge is rather limited :)

**Comment by Bitcoinpunks** [comment]
Thanks! I'll give it a shot. Pretty noobish still here as well. So all tips
are welcome.

**Comment by Bitcoinpunks** [comment]
I could actually deploy to Goerli with metamask.

**Comment by johnnyzen** [comment]
Can anyone test on brave browser / brave wallet?

**Comment by kilogold** [comment]
I had to make an entire eth-brownie project to workaround this issue with Brave Wallet.

I tried using Chrome w/ Metamask and still broken.
I need to use expert dev methods to get my contracts deployed, because Remix IDE just won't work for the same reasons listed on this thread.

tl;dr - my fix was not to use Remix, which is not helpful.

**Comment by yann300** [comment]
@kilogold could you try again using this fix https://deploy-preview-3477--remixproject.netlify.app/
this will probably released tomorrow.
Thanks for your patience.

**Comment by kilogold** [comment]
Same result, @yann300.
That said, Chrome + Metamask is indeed working.
This softens my workaround, but I've personally already got hardhat & brownie projects.

I'm unblocked, but still unable to use Brave for Remix IDE dev.

**Comment by Bitcoinpunks** [comment]
I think so too! 👍🏼

**Comment by romtecmax** [comment]
This was a brave issue and acording to them it has been solved - i didn't double check, but i assume it works now.
[see here](https://community.brave.com/t/internal-json-rpc-error-when-using-brave-wallet-to-deploy-contracts/478192)
